src/Authentication/AuthenticatorFactory.php

```diff
@@ -15,20 +15,17 @@ class AuthenticatorFactory
      */
     public function getAuthenticator(GuzzleClientFactory $clientFactory, $resource)
     {
-        $authenticators = [
-            ClientCredentialsEnvironmentAuthenticator::class,
-            ManagedCredentialsAuthenticator::class,
-        ];
-        foreach ($authenticators as $authenticatorClass) {
-            /** @var AuthenticatorInterface $authenticator */
-            $authenticator = new $authenticatorClass($clientFactory, $resource);
-            try {
-                $authenticator->checkUsability();
-                return $authenticator;
-            } catch (ClientException $e) {
-                $clientFactory->getLogger()->debug($authenticatorClass . ' is not usable: ' . $e->getMessage());
-            }
+        $authenticator = new ClientCredentialsEnvironmentAuthenticator($clientFactory, $resource);
+        try {
+            $authenticator->checkUsability();
+            return $authenticator;
+        } catch (ClientException $e) {
+            $clientFactory->getLogger()->debug(
+                'ClientCredentialsEnvironmentAuthenticator is not usable: ' . $e->getMessage()
+            );
         }
-        throw new ClientException('No suitable authentication method found.');
+        /* ManagedCredentialsAuthenticator checkUsability method has poor performance due to slow responses
+            from GET /metadata requests */
+        return new ManagedCredentialsAuthenticator($clientFactory, $resource);
```

Comment:
V popisu mas "checkUsability method is skipped on instance authorization", nicmene tady se porad `checkUsability` dela, je to OK?

Jestli koukam spravne, tak jediny rozdil, oproti puvodni implementaci je, ze se nedela `checkUsability` na `ManagedCredentialsAuthenticator`? Nebo za tim je jeste neco?

Reply:
hmm.. docetl jsme ted i testy a asi to tam popisujes. Asi ti tady budu verit, ze to dela, co ma, nevim jak ty authenticatory funguji

Reply:
> Jestli koukam spravne, tak jediny rozdil, oproti puvodni implementaci je, ze se nedela checkUsability na ManagedCredentialsAuthenticator? Nebo za tim je jeste neco?

Ano tak. Rozdíl v chování je ten, že předtím:
- zkontrolovalo se jestli jsou k dispozici envy (checkusability na ClientCredentialsEnvironmentAuthenticator)
- pokud se to nezdařilo, tak se zkontrolovalo jestli půjde získat instance credentials (checkusability na ManagedCredentialsAuthenticator)
- pokud se ani to nezdařilo, tak se vyhodila vyjímka

Nově:
- zkontroluje se jestli jsou k dispozici envy (checkusability na ClientCredentialsEnvironmentAuthenticator)
- pokud se to nezdaří, tak se vrátí instance credentials (checkusability na ManagedCredentialsAuthenticator)

Tzn. že praktická změna je v tom, že pokud není k dispozici žádná autentizační metodata, tak to místo toho, že není k dispozici žádná autentizační metodata napíšě, že nelze získat instance credentials.